test/acceptance/acceptance_test.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build !windows
+
+package acceptance
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/acceptance/testrunners"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+	"github.com/stretchr/testify/suite"
+	"testing"
+)
+
+var envMetaDataStrings = &(environment.MetaDataStrings{})
+
+func init() {
+	environment.RegisterEnvironmentMetaDataFlags(envMetaDataStrings)
+}
+
+type AcceptanceTestSuite struct {
+	suite.Suite
+	test_runner.TestSuite
+}
+
+var _ test_runner.ITestSuite = (*AcceptanceTestSuite)(nil)
+
+func (suite *AcceptanceTestSuite) GetSuiteName() string {
+	return "SecurityTestSuite"
+}
+
+var (
+	ec2TestRunners []*test_runner.TestRunner
+)
+
+func getEc2TestRunners() []*test_runner.TestRunner {
+	if ec2TestRunners == nil {
```

Comment:
Why is the `== nil` check needed? What would happen if you just returned a new slice with the FilePermissionTestRunner?

Reply:
Nothing, it just caches the test runners so it doesn't have to be created again. I guess the function isn't pure anymore, so just depends on what design pattern we have agreed on